iroh-net/src/hp/derp/http/mesh_clients.rs

```diff
@@ -0,0 +1,161 @@
+use reqwest::Url;
+use tokio::task::JoinSet;
+use tokio_util::sync::CancellationToken;
+
+use crate::hp::{
+    derp::{http::ClientBuilder, DerpMap, MeshKey, PacketForwarderHandler},
+    key::node::SecretKey,
+};
+
+use super::Client;
+
+/// Spawns, connects, and manages special `derp::http::Clients`.
+///
+/// These clients handled incoming network update notifications from remote
+/// `derp::http::Server`s. These servers are used as `PacketForwarder`s for
+/// peers to which we are not directly connected.
+/// A `mesh_key` is used to ensure the remote server belongs to the same mesh network.
+#[derive(Debug)]
+pub(crate) struct MeshClients {
+    tasks: JoinSet<()>,
+    mesh_key: MeshKey,
+    server_key: SecretKey,
+    mesh_addrs: MeshAddrs,
+    packet_fwd: PacketForwarderHandler<Client>,
+    cancel: CancellationToken,
+}
+
+impl MeshClients {
+    pub(crate) fn new(
+        mesh_key: MeshKey,
+        server_key: SecretKey,
+        mesh_addrs: MeshAddrs,
+        packet_fwd: PacketForwarderHandler<Client>,
+    ) -> Self {
+        Self {
+            tasks: JoinSet::new(),
+            cancel: CancellationToken::new(),
+            mesh_key,
+            server_key,
+            mesh_addrs,
+            packet_fwd,
+        }
+    }
+
+    pub(crate) async fn mesh(&mut self) {
+        let addrs = match &self.mesh_addrs {
+            MeshAddrs::Addrs(urls) => urls.to_owned(),
+            MeshAddrs::DerpMap(derp_map) => {
+                let mut urls = Vec::new();
+                for (_, region) in derp_map.regions.iter() {
+                    for node in region.nodes.iter() {
+                        // note: `node.host_name` is expected to include the scheme
+                        let url: Url = format!("{}/derp", node.host_name).parse().unwrap();
```

Comment:
You can use [`Url::set_path](https://docs.rs/url/latest/url/struct.Url.html#method.set_path)
```rust
let mut url = node.host_name().clone();
url.set_path("/derp");
```